vhdbuilder/packer/install-dependencies.sh

```diff
@@ -363,14 +363,38 @@ for imageToBePulled in ${ContainerImages[*]}; do
     CONTAINER_IMAGE=$(string_replace $downloadURL $version)
     pullContainerImage ${cliTool} ${CONTAINER_IMAGE} &
     image_pids+=($!)
-    echo "  - ${CONTAINER_IMAGE}" >> ${VHD_LOGS_FILEPATH}
+
+    # kube-proxy install's logged after checking for nf_tables
+    if [[ ${CONTAINER_IMAGE} != *"kube-proxy"* ]]; then
+      echo "  - ${CONTAINER_IMAGE}" >> ${VHD_LOGS_FILEPATH}
+    fi
     while [[ $(jobs -p | wc -l) -ge $parallel_container_image_pull_limit ]]; do
       wait -n
     done
   done
 done
 wait ${image_pids[@]}
 
+# this is used by kube-proxy and need to cover previously supported version for VMAS scale up scenario
+# So keeping as many versions as we can - those unsupported version can be removed when we don't have enough space
+# NOTE that we keep multiple files per k8s patch version as kubeproxy version is decided by CCP.
+
+# kube-proxy regular versions >=v1.17.0  hotfixes versions >= 20211009 are 'multi-arch'. All versions in kube-proxy-images.json are 'multi-arch' version now.
+KUBE_PROXY=$(jq '.ContainerImages[] | select(.downloadURL | contains("kube-proxy"))' $COMPONENTS_FILEPATH)
+KUBE_PROXY_IMAGE=$(echo $KUBE_PROXY | jq -r .downloadURL)
+KUBE_PROXY_IMAGE_VERSIONS=$(echo $KUBE_PROXY | jq -r .multiArchVersions)
+for KUBE_PROXY_IMAGE_VERSION in ${KUBE_PROXY_IMAGE_VERSIONS}; do
+  CONTAINER_IMAGE=$(string_replace $KUBE_PROXY_IMAGE $KUBE_PROXY_IMAGE_VERSION)
+  ctr --namespace k8s.io run --rm ${CONTAINER_IMAGE} checkTask /bin/sh -c "iptables --version" | grep -v nf_tables && echo "kube-proxy contains no nf_tables"
```

Comment:
I dont understand what the purpose of this line.  Also why isn't this logic added at the top in the main version loop ?

Reply:
Checking with the team on the purpose. Since it assumes kube-proxy's installed, I am guessing this goes after wait statement

Reply:
I can remove this line, since it is not getting persisted in the vhd log file